.github/workflows/test.yaml

```
@@ -236,14 +236,61 @@ jobs:
           if [ -e "$CACHE_RESULT_STAMP" ]; then
               echo "::set-output name=already-ran::true"
           fi
+    - name: Install jq as a snap
+      if: steps.cached-results.outputs.already-ran != 'true'
+      run: |
+          sudo apt-get remove --purge jq
+          sudo snap install jq
     - name: Run spread tests
       if: "!contains(github.event.pull_request.labels.*.name, 'Skip spread') && steps.cached-results.outputs.already-ran != 'true'"
       env:
           SPREAD_GOOGLE_KEY: ${{ secrets.SPREAD_GOOGLE_KEY }}
       run: |
           # Register a problem matcher to highlight spread failures
           echo "::add-matcher::.github/spread-problem-matcher.json"
-          spread -abend google:${{ matrix.system }}:tests/...
+
+          FAILED_TESTS="${{ github.workspace }}/.test-results/${{ matrix.system }}-failed-tests.outputs.failed-tests"
+          
+          if [ -z "$FAILED_TESTS" ] || [ "$FAILED_TESTS" == "all"] ; then
+              spread -abend google:${{ matrix.system }}:tests/... | tee spread.log
+          else
+              spread -abend "$FAILED_TESTS" | tee spread.log
+          fi
+    - name: analyze spread log
+      run: |
+          echo "Running spread log analyzer"
+          ./tests/lib/tools/log-analyzer spread.log --print-detail none --print-results failed --output spread-results.json
+    - name: spread highlights
+      run : |
+          TESTS_NUMBER=$(spread -list google:${{ matrix.system }}:tests/.. | wc -l)
```

since this is spread, shouldn't this be `...` and not `..` ? 

```suggestion
          TESTS_NUMBER=$(spread -list google:${{ matrix.system }}:tests/... | wc -l)
```

---

Thanks for the comment, I just pushed a simplification for this line.